### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by slilo**

### CODE REVIEW

**Comment by vpetersson** [comment]
Comments: 

* Let's move all test specific packages to `requirements-dev.txt` - no need for them to be installed on regular devices
* Fix the Snyk warnings
* Let's move all tests to 'tests' subfolder(s) to enforce a more clean structure
* I doubt we need this many packages added to `requirements.txt` - let's tighten it up.

**Comment by slilo** [comment]
Closes #10 
Run tests with `pytest --cov=agent`

File: tests/test_agent.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+
+def test_get_device_id(cert):
+    with mock.patch(
+            'builtins.open',
+            mock.mock_open(read_data=cert),
+            create=True
+    ):
+        device_id = agent.get_device_id()
+        assert device_id == 'localhost'
```
Comment (vpetersson) [inline tests/test_agent.py]: s/localhost/foobar.d.wott.local/

File: tests/test_agent.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+
+def test_generate_cert():  # TODO: parse key and csr
+    cert = agent.generate_cert('localhost')
```
Comment (vpetersson) [inline tests/test_agent.py]: s/localhost/foobar.d.wott.local/

File: tests/test_agent.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+    agent.CERT_PATH = str(tmpdir)
+    agent.CLIENT_CERT_PATH = str(crt_path)
+    agent.CLIENT_KEY_PATH = str(key_path)
+    with mock.patch('builtins.print') as prn:
+        with pytest.raises(json.decoder.JSONDecodeError):
+            hello = agent.say_hello()
+        assert mock.call('Hello failed.') in prn.mock_calls
+
+
+def test_uptime(uptime):
```
Comment (vpetersson) [inline tests/test_agent.py]: I don't fully understand this one. Does uptime always end up being 60s when you mock the interface?

File: tests/cassettes/test_send_ping.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+interactions:
```
Comment (vpetersson) [inline tests/cassettes/test_send_ping.yaml]: This is one of the possible states if you have an invalid certificate. If you however do present a proper certificate and auth, you will receive a 200 (or 201 don't remember).

In short, this is a valid mocked result, but only for a "failed" ping so to speak.

File: tests/test_agent.py (hunk 3 above)
Comment (slilo) [reply tests/test_agent.py]: It's pre-populated value.
Please check https://github.com/slilo/agent/blob/tests/tests/conftest.py#L160
It passed in the arguments (pytest magic).

File: tests/cassettes/test_send_ping.yaml (hunk 4 above)
Comment (slilo) [reply tests/cassettes/test_send_ping.yaml]: Updated with proper certificate! Both cases are covered now.

**Pull request merged**